### PR TITLE
indicate HostConfig to be used for "create container"

### LINKED ISSUE #13614: container start accepts `hostConfig` for backwards compatibility
As of https://github.com/docker/docker/commit/1df87b95066198c30312147393c18e0be0564fd0 the `create container` remote api endpoint still accepts a `HostConfig` as POST body, but a [code comment indicates](https://github.com/docker/docker/blob/master/daemon/start.go#L27) that the HostConfig should only be used for `start container`.

The relevant docs produced some confusion (see #13466) after the request body description had been removed and [added](https://github.com/docker/docker/commit/ac12c8053779d36409e248f41c4d80ef0e4fbe9f) again.

I propose to explictly indicate in the docs that the HostConfig is only allowed on `start container` for backwards compatibility.

### PR DESCRIPTION
**Pull request by gesellix**
Closes #13614

Signed-off-by: Tobias Gesellchen tobias@gesellix.de

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks! I'm gonna ask @cpuguy83 to have an extra pair of eyes on this as well. Do we still need a `content-type` header for the request?

ping @cpuguy83 could you have a look at this, just to be sure :)

File: docs/sources/reference/api/docker_remote_api_v1.16.md [hunk 1]
```diff
@@ -504,71 +504,18 @@ Status Codes:
 
 Start the container `id`
 
+> **Note**:
```
Comment (moxiegirl) [inline docs/sources/reference/api/docker_remote_api_v1.16.md]: For backwards compatibility, this endpoint accepts a `HostConfig` as a JSON-encoded request body.

File: docs/sources/reference/api/docker_remote_api_v1.17.md [hunk 2]
```diff
@@ -634,74 +634,18 @@ Status Codes:
 
 Start the container `id`
 
+> **Note**:
+> For backwards compatibility this endpoint accepts a **HostConfig** as JSON encoded request body.
```
Comment (moxiegirl) [inline docs/sources/reference/api/docker_remote_api_v1.17.md]: Ibid. See previous comment.

File: docs/sources/reference/api/docker_remote_api_v1.18.md [hunk 3]
```diff
@@ -675,95 +675,18 @@ Status Codes:
 
 Start the container `id`
 
+> **Note**:
+> For backwards compatibility this endpoint accepts a **HostConfig** as JSON encoded request body.
```
Comment (moxiegirl) [inline docs/sources/reference/api/docker_remote_api_v1.18.md]: Ibid. See previous comment.

File: docs/sources/reference/api/docker_remote_api_v1.19.md [hunk 4]
```diff
@@ -694,93 +694,17 @@ Status Codes:
 
 Start the container `id`
 
+> **Note**:
```
Comment (moxiegirl) [inline docs/sources/reference/api/docker_remote_api_v1.19.md]: Ibid. See previous comment.

File: docs/sources/reference/api/docker_remote_api_v1.20.md [hunk 5]
```diff
@@ -693,93 +693,17 @@ Status Codes:
 
 Start the container `id`
 
+> **Note**:
```
Comment (moxiegirl) [inline docs/sources/reference/api/docker_remote_api_v1.20.md]: Ibid. See previous comment.

**Comment by gesellix** [comment]
@moxiegirl I've added your suggested fixes. When the PR is ready to be merged, I can squash the commits.

**Comment by gesellix** [comment]
@thaJeztah I can verify that the `content-type` header can be omitted when the request body is empty.

**Comment by SvenDowideit** [comment]
LGTM

**Comment by gesellix** [comment]
The commits have been squashed now.

**Comment by moxiegirl** [comment]
LGTM

**Comment by thaJeztah** [comment]
LGTM, thanks, @gesellix!

**Pull request merged**